### PR TITLE
test(partition): add marble tests for partition operator

### PR DESCRIPTION
**Pull request by kwonoj**
relates to #309 

Replace existing test to marble test, migrating RxJS4 test also.

### CODE REVIEW

File: spec/operators/partition-spec.js [hunk 1]
```diff
@@ -1,78 +1,81 @@
...
+    var unsub =     '-------!';
+    var expected = ['--a-',
+                    '----b-'];
+    var result = e1.partition(predicate);
 
-    positiveStream.subscribe(rejecter, completer, rejecter);
-    negativeStream.subscribe(rejecter, completer, rejecter);
+    for (var idx = 0; idx < result.length; idx++ ) {
+      expectObservable(result[idx], unsub).toBe(expected[idx]);
+    }
```
Comment (staltz) [inline spec/operators/partition-spec.js]: Maybe this one too should use `expectObservableArray` ?

Comment (kwonoj) [inline spec/operators/partition-spec.js]: This test case is only case accepting unsubscription as parameter, choose simplify `expectObservableArray` and let loop explicitly with unsubscription.

Comment (staltz) [inline spec/operators/partition-spec.js]: alright

**Comment by blesh** [comment]
We still have a few missing tests for this operator. It looks like this was meant to just rewrite the existing tests, would you like to tackle the rest @kwonoj?

The ones I can think of that are missing are:

- empty
- throw
- error thrown in predicate
- never
- predicate never matches anything
- predicate always matches everything
- no predicate passed (should error I think?)

**Comment by kwonoj** [comment]
Absolutely. As mentioned this PR was intended to 'migrate' only, but it makes more sense to add additional coverage. I'll cover them up and update PR.

**Comment by kwonoj** [comment]
Updated PR by adding additional test coverage.

**Comment by blesh** [comment]
merged with 0fcf52e... Thanks @kwonoj!

**Pull request closed without merging**